### ISSUE TITLE
Crash when `AssignAttr` is inferred as a class

### ISSUE THREAD

**Issue by areveny** [PyCQA/astroid#1374]
### Steps to reproduce

Thorough initial triage can be found here:

https://github.com/PyCQA/astroid/pull/1373

Basically with `Django` and a construction like this that reassigns a module attribute:

```
import datetime

# Change the modules
datetime.date = FakeDate
```

The `AssignAttr` can be inferred where we normally only expect `ClassDef`, resulting in an error when we try to get the `qname` of the inferred node.

### Current behavior
`AssignAttr` gets inferred, and a crash results.

### Expected behavior

No crash, whether we decide to make `AssignAttr` result from the inference or not.

Resolving this resolves https://github.com/PyCQA/pylint/issues/5717

`2.10.0-dev0`

**Comment by DanielNoord** [comment]
@areveny Do you want to investigate this? I'll be stuck on a plane for two hours tonight so I could also try and take a look?

**Comment by areveny** [comment]
Yup, looking at this. I think your experience would be very helpful. I agree there could be a root cause in producing the `AssignAttr` before we try to infer on it.

**Comment by DanielNoord** [comment]
I'm not sure what the best approach is here, but it feels like the issues are related. The last reproducing code in the previous PR I posted has the benefit of not requiring any dependencies. The `django` file can be empty as long as the imported class is defined (with an ellipsis as body).
That might help troubleshooting this.


----------

I'm not sure what your debug setup is, so perhaps you found something that works but what I do is:
```python
from pylint.lint import Run

Run(["test.py"])
```
And then have the offending code in `test.py`. I use VS Code, so when I run this file I can add breakpoints in the `astroid` codebase and actually see what is going on. (This assumes you're using a editable install from `pip install -r .`.

**Comment by areveny** [comment]
Resolved by the merged pylint-django issue: https://github.com/PyCQA/pylint-django/issues/349

**Comment by DanielNoord** [comment]
@areveny Was this comment fixed as well: https://github.com/PyCQA/astroid/pull/1373#issuecomment-1023987021?

I think we identified two issues while working on this and I wonder if the latter has been fixed.

**Comment by areveny** [comment]
Logically yes, I believe providing an `AssignAttr` where we expect a `ClassDef` is the root cause of these `qname` errors and this was fixed.

But I deleted my old local environment and am having a bit of trouble reproducing your error on `pylint-django==2.5.0`. (I was able to reproduce it before when I tried to install the original reporter's `license-manger` project.) If we want some further assurance I could look into it more.
